src/GPX.jl

```diff
@@ -7,6 +7,13 @@ import Geodesy: LLA
 import TimeZones: astimezone, localzone
 import PooledArrays: PooledArray
 
+function gpx_segment2pts(segment; filename=filename, tz=localzone())
+    p = segment.points
+    df = DataFrame(time=[astimezone(p1.time, tz) for p1 ∈ p], location=[LLA(p1.lat, p1.lon) for p1 ∈ p])
+    df.filename = repeat(PooledArray([filename]), size(df, 1))
+    df
+end
+
 function read(filename; tz=localzone())
```

Comment:
May I suggest that we always return a `DataFrame` and simply add a field to the `DataFrame` to tell which `segment` the point is from? Much cleaner.

Reply:
```
combine_segments(gpx_segments) = vcat([insertcols!(x, :segment_index=>i) for (i,x) in enumerate(gpx_segments)]...)
```

alright, added additional column :segment_index if there are multiple segments.
So it will return Datafarme with an extra column